File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    let arr=Array(n+1).fill(-1);
    arr[1]=1;
    arr[2]=2;
    arr[3]=3;
    return solve(n,arr)
};

function solve(n,arr){
    if(arr[n]==-1){
        arr[n]=solve(n-1,arr)+solve(n-2,arr);
    }
    return arr[n];
}

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int[]dp=new int[n+1];
        return solve(n,dp);
    }
    public int solve(int n,int[] dp){
        if(n==1 || n==2){
            return n;
        }
        if(dp[n]==0){
            dp[n]=solve(n-1,dp)+solve(n-2,dp);
        }
        return dp[n];
    }
}